### PR TITLE
Dev 3187 setup profiling

### PR DESCRIPTION
**Pull request by harsha09**
#### What's this PR do?
Setup profiling in sentry.

#### Why are we doing this? How does it help us?
Helps to identify performance bottlenecks and address them.

#### How should this be manually tested? Please include detailed step-by-step instructions.
Login to https://news-revenue-hub.sentry.io/profiling/?project=6046263, profiling should be enabled.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?
NA

#### Have automated unit tests been added? If not, why?
NA

#### How should this change be communicated to end users?
NA

#### Are there any smells or added technical debt to note?
NA

#### Has this been documented? If so, where?
NA

#### What are the relevant tickets? Add a link to any relevant ones.
[DEV-3187]

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?
Added env variable `SENTRY_PROFILING_SAMPLE_RATE` takes float values from 0 - 1.0 represents [0% - 100%].
The value defaults to 1.0.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:
NA

[DEV-3187]: https://news-revenue-hub.atlassian.net/browse/DEV-3187?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: revengine/settings/deploy.py [hunk 1]
```diff
@@ -132,6 +132,7 @@
         environment=ENVIRONMENT,
         # https://docs.sentry.io/platforms/python/configuration/sampling/#setting-a-uniform-sample-rate
         traces_sample_rate=1.0,  # TODO: DEV-2683 After testing will want to reduce this to less than 100% sampling rate.
+        _experiments={"profiles_sample_rate": SENTRY_PROFILING_SAMPLE_RATE},
```
Comment (x110dc) [inline revengine/settings/deploy.py:135]: This is the syntax for an older version of the Sentry package. Let's just upgrade to the latest (1.20.0 as of right now) and use the non-experimental syntax.

**Review by x110dc — CHANGES_REQUESTED** [review]
Also, can you research and provide some guidance about what we set `SENTRY_PROFILING_SAMPLE_RATE` to in production? Thanks.

File: revengine/settings/deploy.py (hunk 1 above)
Comment (harsha09) [reply revengine/settings/deploy.py]: Addressed with 0bc73cbf7377f1df9538f960c10b59b4f5341746.

File: revengine/settings/base.py [hunk 2]
```diff
@@ -300,6 +300,7 @@ class CurrencyDict(TypedDict):
 SENTRY_ENABLE_BACKEND = os.getenv("SENTRY_ENABLE_BACKEND", "false").lower() == "true"
 SENTRY_DSN_BACKEND = os.getenv("SENTRY_DSN_BACKEND")
 SENTRY_ENABLE_PII = os.getenv("SENTRY_ENABLE_PII", "true").lower() == "true"
+SENTRY_PROFILING_SAMPLE_RATE = float(os.getenv("SENTRY_PROFILING_SAMPLE_RATE", "1.0"))
```
Comment (x110dc) [inline revengine/settings/base.py:303]: Let's set the default rate to something that would be safe in production, so may "0.1". Then good to go.

**Review by x110dc — CHANGES_REQUESTED** [review]

**Review by x110dc — APPROVED** [review]

**Pull request merged**